src/rules/stylelint-disable-reason/index.js

```diff
@@ -0,0 +1,59 @@
+import {
+  report,
+  ruleMessages,
+  validateOptions,
+} from "../../utils"
+
+export const ruleName = "stylelint-disable-reason"
+
+export const messages = ruleMessages(ruleName, {
+  expectedPreceding: "Expected comment reason preceding stylelint disable comment",
+  expectedSucceeding: "Expected comment reason succeeding stylelint disable comment",
+})
+
+const stylelintDisableCommand= "stylelint-disable"
+
+export default function (expectation) {
+  return function (root, result) {
+    result.stylelint = result.stylelint || {}
+    result.stylelint.disabledRanges = false
+
+    const validOptions = validateOptions(result, ruleName, {
+      actual: expectation,
+      possible: [
+        "always-preceding",
+        "always-succeeding",
+      ],
+    })
+    if (!validOptions) { return }
+
+    root.walkComments(function (comment) {
+      if (comment.text.indexOf(stylelintDisableCommand) === 0) {
```

Comment:
Please change this to an early return to avoid unnecessary indentation and blocking: `if (comment.text.indexOf(stylelintDisableCommand) !== 0) { return }`